### PR TITLE
Add humidifier entity row

### PR DESCRIPTION
**Pull request by Shulyaka**
## Proposed change

Add entity row for humidifier devices.

entities

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
entities:
  - entity: humidifier.humidifier
  - entity: humidifier.dehumidifier
  - entity: humidifier.hygrostat
  - entity: climate.hvac
  - entity: climate.heatpump
  - entity: climate.ecobee
  - entity: cover.kitchen_window
show_header_toggle: true
title: Humidifier comparison
type: entities
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by zsarnett** [comment]
I feel like this should be more similar to a Climate Row. The top section is the toggle and under the toggle `Currently: 25%`

**Comment by Shulyaka** [comment]
We don't have the current humidity for the humidifier entity, only target.

**Comment by zsarnett** [comment]
Okay. Then change currently to target

**Comment by bramkragten** [comment]
> We don't have the current humidity for the humidifier entity, only target.

I'm not familiar with humidifiers, but it sounds logical that some will provide the current humidity? As you can set a target it will know what the current is?

**Comment by Shulyaka** [comment]
Well, it was decided that the current humidity must be a separate entity (https://github.com/home-assistant/architecture/issues/310#issuecomment-627035435). So the target humidity, mode and status are all we have.
I also don't think we should move the toggle because firstly it occupies the full height and secondly if we change that, and the user adds other togglable entities on the same card, it would look out of place.

**Comment by bramkragten** [comment]
Do we have a reference to this separate entity? In the frontend we should still show them together I think.

**Comment by Shulyaka** [comment]
They could be linked via the device registry. Is it available i frontend?
And we also need to consider a case where it is not linked.

**Comment by zsarnett** [comment]
Lets just move the Current Humidity to the secondary info location. This will be similar to how we allow the current brightness in the secondary info. This wont be an option to choose but set via this row type. Let get this PR merged for 0.116

**Comment by Shulyaka** [comment]
I am sorry but unfortunately couldn't quite get the idea.
What is the secondary info location? If you mean the "more info dialog", then it is already there as a slider even without this PR:
humidifier-more-info

**Comment by zsarnett** [comment]
No

In the entity row. 
image

**Comment by Shulyaka** [comment]
I couldn't find an example how current brightness is used in the secondary info. May I ask you to point me to a right direction?)

File: src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      `;
+    }
+
+    return html`
+      <hui-generic-entity-row .hass=${this.hass} .config=${this._config}>
+        <ha-entity-humidifier
+          .hass=${this.hass}
+          .stateObj=${stateObj}
+        ></ha-entity-humidifier>
+      </hui-generic-entity-row>
```
Comment (zsarnett) [inline src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts:57]:
```suggestion
      <hui-generic-entity-row .hass=${this.hass} .config=${this._config} .secondaryText=${"Current Humidity: 28%"}>
        <ha-entity-humidifier
          .hass=${this.hass}
          .stateObj=${stateObj}
        ></ha-entity-humidifier>
      </hui-generic-entity-row>
```

Comment (zsarnett) [reply src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts:57]: And then you also wont need the other file. Just insert the toggle into the generic row

Comment (Shulyaka) [reply src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts:57]: Ok, will do that.

Comment (Shulyaka) [reply src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts]: Done, please check

humidifier_comparison

File: src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      <hui-generic-entity-row
+        .hass=${this.hass}
+        .config=${this._config}
+        .secondaryText=${stateObj.attributes.humidity
+          ? html`${this.hass!.localize("ui.card.humidifier.humidity")}:
+            ${stateObj.attributes.humidity}%${stateObj.attributes.mode
+              ? html` (${this.hass!.localize(
+                  `state_attributes.humidifier.mode.${stateObj.attributes.mode}`
+                ) || stateObj.attributes.mode})`
+              : ""}`
```
Comment (zsarnett) [inline src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts:61]: this should be a string not a Template result. Just remove the `html` prefix

File: src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      ha-humidifier-state {
+        text-align: right;
+      }
+    `;
+  }
+}
```
Comment (zsarnett) [inline src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts:79]: note used

**Review by zsarnett — APPROVED** [review]

File: src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import {
+  css,
+  CSSResult,
```
Comment (zsarnett) [inline src/panels/lovelace/entity-rows/hui-humidifier-entity-row.ts:3]:
```suggestion
```

**Review by zsarnett — APPROVED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**